### PR TITLE
Authenticate the sharing request

### PR DESCRIPTION
**Pull request by Gara64**
The requests to the Data-System must be authenticated

### CODE REVIEW

File: server/share/share_handler.coffee [hunk 1]
```diff
@@ -2,7 +2,10 @@ cozydb    = require 'cozydb'
 Client    = require('request-json').JsonClient
 urlHelper = require 'cozy-url-sdk'
 
-
+client = new Client urlHelper.dataSystem.url()
```
Comment (gregorylegarec) [inline server/share/share_handler.coffee]: Why extract `client` from the function ? Always avoid to rely on out-of-scope variables in functions.

File: server/share/share_handler.coffee [hunk 2]
```diff
@@ -2,7 +2,10 @@ cozydb    = require 'cozydb'
...
+if process.env.NODE_ENV is "production" or process.env.NODE_ENV is "test"
```
Comment (gregorylegarec) [inline server/share/share_handler.coffee]: Is this line mandatory ? I am not a big fan of testing explicitly the environment at this stage of the code.

File: server/share/share_handler.coffee [hunk 3]
```diff
@@ -30,7 +33,7 @@ module.exports.sendShareInvitations = (event, callback) ->
             needSaving   = true
 
     # Send the request to the datasystem
-    client = new Client urlHelper.dataSystem.url()
+
```
Comment (gregorylegarec) [inline server/share/share_handler.coffee]: empty line

**Comment by Gara64** [comment]
This should not be merged. The new version of cozy-db will make this PR obsolete.

**Pull request closed without merging**